stage0/src/main.rs

```diff
@@ -212,14 +202,22 @@ pub extern "C" fn rust64_start(encrypted: u64) -> ! {
         SS::set_reg(ds);
     }
 
-    let idt = unsafe { BOOT_IDT.write(InterruptDescriptorTable::new()) };
+    let idt = alloc
+        .leak(InterruptDescriptorTable::new())
+        .expect("Failed to allocate memory for IDT");
 
     create_idt(idt);
     idt.load();
 
-    let pml4 = unsafe { BOOT_PML4.write(PageTable::new()) };
-    let pdpt = unsafe { BOOT_PDPT.write(PageTable::new()) };
-    let pd = unsafe { BOOT_PD.write(PageTable::new()) };
+    let pml4 = alloc
+        .leak(PageTable::new())
```

Comment:
This is not a memory leak right? =)

Reply:
In the strictest sense of the word, it is :)

Namely, once you call `leak`, the memory is never deallocated (as you'd expect when something goes out of scope). That'd make it a memory leak, but that's exactly what we want: we want the data structures to stick around in memory.

You can get the memory back if you call `BOOT_ALLOC.reset()`, which causes all allocations to become invalid, so in that sense it's not a proper memory leak as the memocy _can_ be recovered. But we don't want to; we explicitly want to leak the memory.